src/lnp/application/payment/types.rs

```diff
@@ -256,3 +261,98 @@ impl DumbDefault for TempChannelId {
         Self(Default::default())
     }
 }
+
+/// Lightning network short channel Id as per BIP7
+#[cfg_attr(feature = "serde", serde_as(as = "DisplayFromStr"))]
```

Comment:
It happened that top-level `serde_as(as = ...)` do not work. You need to check how I solved this issue in https://github.com/LNP-BP/rust-lnpbp/commit/cd257be402cbcb5681b8aa1a503f5a5af69f1ba8

Here, specifically, you need to remove all serde derives (and `serde_as`) and use that on the upstream structures containing this `ShortChannelId`

Reply:
Noted.

Reply:
What if I remove `serde_as` for `ShortChannelId`? Is it required here?

Reply:
You can remove anything with serde here, including serde_as